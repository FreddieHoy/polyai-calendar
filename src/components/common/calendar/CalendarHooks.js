import React, { useState, useEffect } from 'react';
import styled from 'styled-components';
import WeekDays from './WeekDays';
import Dates from './Dates';
import MonthYear from './MonthYear';
import {
  getYear,
  getMonthName,
  getDatesArray,
  getHighlightedDay
} from './Helper';

const Container = styled.div`
  border: 1px solid black;
  width: 352px;
  text-align: center;
`;

const CalendarHooks = () => {
  const [targetDate, setTargetDate] = useState(new Date());
  const [month, setMonth] = useState('');
  const [year, setYear] = useState('');
  const [today, setToday] = useState('');
  const [datesArray, setDatesArray] = useState([]);

  const updateVariables = (targetDate) => {
    setYear(getYear(targetDate));
    setMonth(getMonthName(targetDate));
    setDatesArray(getDatesArray(targetDate));
    setToday(getHighlightedDay(targetDate));
  };

  const switchMonth = (e) => {
    const nextMonthIndexNumber = targetDate.getMonth() + (e.target.id ? 1 : -1);
    const adjacentMonth = new Date(targetDate.setMonth(nextMonthIndexNumber));

    setTargetDate(adjacentMonth);
    updateVariables(targetDate);
  };

  useEffect(() => {
    updateVariables(targetDate);
  }, [targetDate]);

  return (
    <Container>
      <MonthYear month={month} year={year} switchMonth={switchMonth} />
      <WeekDays />
      <Dates array={datesArray} today={today} />
    </Container>
  );
};

export default CalendarHooks;
