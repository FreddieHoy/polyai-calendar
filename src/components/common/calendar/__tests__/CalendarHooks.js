import { shallow } from 'enzyme';
import React from 'react';

import CalendarHooks from '../CalendarHooks';

describe('CalendarHooks', () => {
  it('renders the correct components', () => {
    const wrapper = shallow(<CalendarHooks />);
    expect(wrapper).toMatchSnapshot();
  });
});
