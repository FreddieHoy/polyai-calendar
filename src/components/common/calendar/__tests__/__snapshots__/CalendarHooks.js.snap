// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarHooks renders the correct components 1`] = `
<styled.div>
  <MonthYear
    month=""
    switchMonth={[Function]}
    year=""
  />
  <WeekDays />
  <Dates
    array={Array []}
    today=""
  />
</styled.div>
`;
